opengrok-indexer/src/main/java/org/opengrok/indexer/analysis/TextAnalyzer.java

```diff
@@ -91,10 +91,13 @@ public Xrefer writeXref(WriteXrefArgs args) throws IOException {
      */
     protected abstract Xrefer newXref(Reader reader);
 
+    /**
+     * Gets a BOM-stripped {@link Reader} (default UTF-8 charset) of the
+     * specified {@code stream}, wrapped in a {@link ZeroReader}.
+     */
     protected Reader getReader(InputStream stream) throws IOException {
-        // SRCROOT is read with UTF-8 as a default.
-        return IOUtils.createBOMStrippedReader(stream,
-            StandardCharsets.UTF_8.name());
+        // sourceRoot is read with UTF-8 as a default.
+        return new ZeroReader(IOUtils.createBOMStrippedReader(stream,
```

Comment:
why is this suddenly needed ? What Reader suffers from the 0 return problem specifically ?

Reply:
JFlex just advises that on some JRE implementations, they found readers returning 0 sometimes. JFlex prior to 1.6.1 did a guard against this, but afterward JFlex offered `ZeroReader` as the guard, which needs to be explicitly used. I'm putting `ZeroReader` then into OpenGrok, since the workaround is desirable.